src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -274,6 +275,42 @@ object Simplify {
       case MatrixColsTable(MatrixFilterRows(child, _)) => MatrixColsTable(child)
       case MatrixColsTable(MatrixAggregateRowsByKey(child, _)) => MatrixColsTable(child)
 
+      case TableHead(TableMapRows(child, newRow, newKey, preservedKeyFields), n) =>
+        TableMapRows(TableHead(child, n), newRow, newKey, preservedKeyFields)
+      case TableHead(TableRepartition(child, nPar, shuffle), n) =>
+        TableRepartition(TableHead(child, n), nPar, shuffle)
```

Comment:
I think the number of partitions should be adjusted. Doing this results in 2 partitions not 10 partitions.
```
table = hl.utils.range_table(10)
In [9]: table.repartition(n=10).head(3).n_partitions()
Out[9]: 2
```

Also, the answer that repartition and head is returning for the number of partitions is one off from what I would have expected. 9 instead of 10 and 2 instead of 3 respectively.

Reply:
I don't think this is a problem -- the `repartition` is going to use a hash partitioner which produces imbalanced partitions, and `head` will then take as many as are needed to produce 3 rows. If the hash partitioner shuffled 2 rows into either partition 1 or 2, then this is consistent

Reply:
the `9` actually confirms this.

Reply:
(repartition with shuffle=True does a hash shuffle, then a coerce. empty partitions are dropped at that point, dropping the empty one)